src/models/lightfm.py

```diff
@@ -0,0 +1,243 @@
+        prediction difference between a positive example and a randomly
+        chosen negative example. Useful when only positive interactions
+        are present and optimising ROC AUC is desired.
+        - WARP: Weighted Approximate-Rank Pairwise [2]_ loss. Maximises
+        the rank of positive examples by repeatedly sampling negative
+        examples until rank violating one is found. Useful when only
+        positive interactions are present and optimising the top of
+        the recommendation list (precision@k) is desired.
+        - k-OS WARP: k-th order statistic loss [3]_. A modification of WARP that
+        uses the k-th positive example for any given user as a basis for pairwise
+        updates.
+
+        Two learning rate schedules are available:
+
+        - adagrad: [4]_
+        - adadelta: [5]_
+
+        Parameters
+        ----------
+
+        no_components: int, optional
+            the dimensionality of the feature latent embeddings.
+        k: int, optional
+            for k-OS training, the k-th positive example will be selected from the
+            n positive examples sampled for every user.
+        n: int, optional
+            for k-OS training, maximum number of positives sampled for each update.
+        learning_schedule: string, optional
+            one of ('adagrad', 'adadelta').
+        loss: string, optional
+            one of  ('logistic', 'bpr', 'warp', 'warp-kos'): the loss function.
+        learning_rate: float, optional
+            initial learning rate for the adagrad learning schedule.
+        rho: float, optional
+            moving average coefficient for the adadelta learning schedule.
+        epsilon: float, optional
+            conditioning parameter for the adadelta learning schedule.
+        item_alpha: float, optional
+            L2 penalty on item features. Tip: setting this number too high can slow
+            down training. One good way to check is if the final weights in the
+            embeddings turned out to be mostly zero. The same idea applies to
+            the user_alpha parameter.
+        user_alpha: float, optional
+            L2 penalty on user features.
+        max_sampled: int, optional
+            maximum number of negative samples used during WARP fitting.
+            It requires a lot of sampling to find negative triplets for users that
+            are already well represented by the model; this can lead to very long
+            training times and overfitting. Setting this to a higher number will
+            generally lead to longer training times, but may in some cases improve
+            accuracy.
+        random_state: int seed, RandomState instance, or None
+            The seed of the pseudo random number generator to use when shuffling
+            the data and initializing the parameters.
+
+        epochs: (int, optional) number of epochs to run
+
+        event_weights_multiplier: int, optional
```

Comment:
I think this param can be removed since it does not affect the final results.
I received exactly the same evaluation metrics for event_weights_multiplier=0.01 and for event_weights_multiplier=100